pkg/controller/mq/broker/setup.go

```diff
@@ -0,0 +1,92 @@
+package broker
+
+import (
+	"context"
+	"time"
+
+	"k8s.io/client-go/util/workqueue"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+
+	svcsdk "github.com/aws/aws-sdk-go/service/mq"
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/crossplane/crossplane-runtime/pkg/event"
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	"github.com/crossplane/crossplane-runtime/pkg/meta"
+	"github.com/crossplane/crossplane-runtime/pkg/ratelimiter"
+	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+
+	svcapitypes "github.com/crossplane/provider-aws/apis/mq/v1alpha1"
+	awsclients "github.com/crossplane/provider-aws/pkg/clients"
+)
+
+// SetupBroker adds a controller that reconciles Broker.
+func SetupBroker(mgr ctrl.Manager, l logging.Logger, rl workqueue.RateLimiter, poll time.Duration) error {
+	name := managed.ControllerName(svcapitypes.BrokerGroupKind)
+	opts := []option{
+		func(e *external) {
+			e.postCreate = postCreate
+			e.preObserve = preObserve
+			e.preUpdate = preUpdate
+			e.preDelete = preDelete
+			e.postObserve = postObserve
+		},
+	}
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		WithOptions(controller.Options{
+			RateLimiter: ratelimiter.NewDefaultManagedRateLimiter(rl),
+		}).
+		For(&svcapitypes.Broker{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(svcapitypes.BrokerGroupVersionKind),
+			managed.WithInitializers(managed.NewDefaultProviderConfig(mgr.GetClient())),
+			managed.WithExternalConnecter(&connector{kube: mgr.GetClient(), opts: opts}),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
```

Comment:
Looks like passing poll interval did not get included in the rebase:
```suggestion
		Complete(managed.NewReconciler(mgr,
			resource.ManagedKind(svcapitypes.BrokerGroupVersionKind),
			managed.WithInitializers(managed.NewDefaultProviderConfig(mgr.GetClient())),
			managed.WithExternalConnecter(&connector{kube: mgr.GetClient(), opts: opts}),
			managed.WithPollInterval(poll),
			managed.WithLogger(l.WithValues("controller", name)),
			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
```

Reply:
what should be done to get this resolved? should I try rebasing it again?

Reply:
@praveenghuge nope, just adding it in :)

Reply:
added the change